dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/ingest/tabulardata/impl/plugins/csv/CSVFileReader.java

```diff
@@ -195,7 +196,7 @@ public int readFile(BufferedReader csvReader, DataTable dataTable, PrintWriter f
                     List<String> args = Arrays.asList("" + (parser.getCurrentLineNumber() - 1),
                                                       "" + headers.size(),
                                                       "" + record.size());
-                    throw new IOException(BundleUtil.getStringFromBundle("ingest.csv.recordMismatch", args));
+                    throw new IngestException(IngestError.RECORD_MISMATCH, args.toArray(new String[0]));
```

Comment:
Consider a constructor accepting a `List` instead of calling `toArray` and than building a list from that array in the constructor.

Reply:
If I would accept `List` then in a lot of method calls I would have to wrap arguments with `Arrays.asList()`. So in the end it wouldn't change much.

Reply:
Why not have both? One accepting a `List` and another with varags converted to a `List`?

Reply:
Hmm, I felt like wrapping stuff wasn't a big deal, but creating additional constructor also isn't, so now that I think about it I don't see why not.